Draft-Accepted/RFCXXXX-Command-Shell-Integration.md

```diff
@@ -0,0 +1,326 @@
+---
+RFC: RFCNNNN
+Author: Steve Lee
+Status: Draft
+SupercededBy: NA
+Version: 1.0
+Area: Interactive Shell
+Comments Due: June 1, 2023
+Plan to implement: Yes
+---
+
+# Command Shell Integration
+
+Executables can provide a more integrated experience with a particular shell by indicating features it supports which the shell can leverage.
+
+Shell integration feature areas:
+
+- Structured output and formatting
+- Tab completion
+- Help
+- Predictors (PowerShell)
+- Feedback Providers (PowerShell)
+
+Traditionally, POSIX shells have pre-defined filesystem locations where tools would put their tab completion scripts or help content.
+However, this typically works for the system and not for a particular user who doesn't have administrative rights to the system.
+This can also make it more complicated to uninstall tools that place files across the filesystem.
+
+Although the examples here are focused on PowerShell, the design is intended to be generic and can be used by other shells.
+However, PowerShell specific implementation details are included.
+
+## Motivation
+
+As a tool developer,
+I can easily integrate with a shell,
+so that users are more productive having a consistent experience within that shell.
+
+## Current Experience
+
+Some tools that are executables and not modules/cmdlets have already reached out on how they can have better integration with PowerShell.
+The current experience requires the user to know about the integration scripts and add them to their `$profile` so that it gets loaded.
+This greatly decreases usage and thus motivation for tooling to produce integration scripts due to lack of discovery and also
+manual effort by the user.
+
+## Enhanced User Experience
```

Comment:
user should be made aware a new item was added and the location.  If something is being loaded, the user should be made aware.  I get people want silence but if a one time shoulder tap should be there so the user is made aware

Reply:
It would make sense that unless `-nologo` is used, we should emit any auto-loaded integration.  Also, any integration is only autoloaded in interactive mode.

Reply:
adding some text on this